sklearn/linear_model/_ridge_solvers.py

```diff
@@ -0,0 +1,221 @@
+import warnings
+
+import numpy as np
+from scipy import linalg
+from scipy import sparse
+
+from ..utils.extmath import safe_sparse_dot
+from ..exceptions import ConvergenceWarning
+from scipy.sparse import linalg as sp_linalg
+
+
+def _cholesky_helper(X, y, alpha, n_features, n_samples):
+    if n_features > n_samples:
+        K = safe_sparse_dot(X, X.T, dense_output=True)
+        try:
+            dual_coef = _solve_cholesky_kernel(K, y, alpha)
+
+            coef = safe_sparse_dot(X.T, dual_coef, dense_output=True).T
+        except linalg.LinAlgError:
+            # use SVD solver if matrix is singular
+            coef = _solve_svd(X, y, alpha)
```

Comment:
Is it ever possible for there to be a `LinAlgError` here? If `n_features > n_samples`, then I think the matrix cannot be singular. I'm having issues with test coverage because codecov is treating this like a new / untracked file, and I don't think a test could ever hit this exception. 

cc @amueller

Reply:
check if coverage was already missing in master